Added/Edited new chapter/assembly about foreman webhooks (#612)
theforeman/foreman-documentation: guides/common/modules/proc_creating-a-webhook.adoc

[id="creating-a-webhook_{context}"]
= Creating a Webhook

> 

Reviewer: Modules would always have some context.

.Procedure

> . In the {Project} web UI, click *Administer* > *Webhooks*.

Reviewer 2: ```suggestion
. In the {ProjectWebUI}, navigate to *Administer* > *Webhooks*.
```

. Click *Create Webhook*.
. Click *Subscribe to field* to reveal a list of available events.
. Select an event from the list.
. Enter a name.

> . Enter a target URL. When using the shellhooks plugin, the URL should be in the form of `https://{smartproxy.example.com}:{smartproxy_port}/shellhooks/my_script`.

Reviewer: Please use one sentence per line.

> . Enter a target URL. When using the shellhooks plugin, the URL should be in the form of `https://{smartproxy.example.com}:{smartproxy_port}/shellhooks/my_script`.

Reviewer 2: ```suggestion
. Enter a target URL.
When using the shellhooks plugin, the URL should be in the form of `https://{smartproxy-example-com}:{smartproxy_port}/shellhooks/my_script`.
```

one sentence per line and the attribute uses dashes, not dots. See [base attributes L95](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-base.adoc#L95)

. Click *Template* to see the list of available templates.
. Select a template from the list.
. Enter a HTTP method.
. Ensure that the *Enabled* flag is checked if you want to create active webhook.

> In the Credentials tab:

Reviewer: We do not document by tab. 
Please rewrite with the step "Click the Credentials tab" and then integrate the rest of the steps as steps on their own. 
We write user-oriented not web ui oriented.

> . Optional: Enter the username and password when HTTP authentication is required. Do not enter anything when using shellhooks.

Reviewer: Please use one sentence per line, as is our agreed style.

> . Select *Verify SSL* if the server certificate should be verified against the system certificate store or Foreman CA. When using shellhooks, ensure this flag is selected.

Reviewer: Please use one sentence per line. As per our agreed style. This will not effect the formatting.

. Select *Proxy Authorization* when using shellhooks, otherwise deselect this box.

> . Optional: Provide one or more CA certificates in the X509 Certification Authorities. Keep this field blank when using shellhooks.

Reviewer: Please use one sentence per line.

> In the Additional tab:

Reviewer: Same here

. Enter the HTTP Content Type, typically application/json, application/xml or text/plain depending on the payload.

> . Optional: Provide HTTP headers as JSON. ERB is allowed and the same macros and variables are available as in webhook templates.

Reviewer: Please use one sentence per line.

> When configuring webhooks with endpoints with non-standard HTTP(s) ports, SELinux port must be assigned, see  https://access.redhat.com/documentation/en-us/red_hat_satellite/6.9/html-single/installing_satellite_server_from_a_connected_network/index#configuring-selinux-to-ensure-access-on-custom-ports_satellite[Configuring SELinux to Ensure Access to Satellite on Custom Ports] in _Installing Satellite Server From A Connected Network_.

Reviewer: We do not use (s) in IBM style: 

IBM style page 241:

Do not form a plural by adding (s). Try to rewrite the sentence to use either the plural
form or singular form, whichever is more appropriate. If you must indicate both forms,
repeat the noun, or use one or more. For example, instead of writing “enter the registra-
tion number(s),” write either “enter the registration number or numbers” or “enter one or
more registration numbers.”

> When configuring webhooks with endpoints with non-standard HTTP(s) ports, SELinux port must be assigned, see  https://access.redhat.com/documentation/en-us/red_hat_satellite/6.9/html-single/installing_satellite_server_from_a_connected_network/index#configuring-selinux-to-ensure-access-on-custom-ports_satellite[Configuring SELinux to Ensure Access to Satellite on Custom Ports] in _Installing Satellite Server From A Connected Network_.

Reviewer: Please use the correct attributes for the links and test them locally.

Reviewer: Please do not use Satellite Server. Use the attributes.

> When configuring webhooks with endpoints with non-standard HTTP(s) ports, SELinux port must be assigned, see  https://access.redhat.com/documentation/en-us/red_hat_satellite/6.9/html-single/installing_satellite_server_from_a_connected_network/index#configuring-selinux-to-ensure-access-on-custom-ports_satellite[Configuring SELinux to Ensure Access to Satellite on Custom Ports] in _Installing Satellite Server From A Connected Network_.

Reviewer: Please use one sentence per line.

> When configuring webhooks with endpoints with non-standard HTTP(s) ports, SELinux port must be assigned, see  https://access.redhat.com/documentation/en-us/red_hat_satellite/6.9/html-single/installing_satellite_server_from_a_connected_network/index#configuring-selinux-to-ensure-access-on-custom-ports_satellite[Configuring SELinux to Ensure Access to Satellite on Custom Ports] in _Installing Satellite Server From A Connected Network_.

Reviewer 2: ```suggestion
When configuring webhooks with endpoints with non-standard HTTP(s) ports, SELinux port must be assigned, see  https://access.redhat.com/documentation/en-us/red_hat_satellite/{AccessRedHatComVersion}/html-single/installing_satellite_server_from_a_connected_network/index#configuring-selinux-to-ensure-access-on-custom-ports_satellite[Configuring SELinux to Ensure Access to Satellite on Custom Ports] in _Installing Satellite Server From A Connected Network_.
```

Additionally, use an asciidoc attribute to link to access.redhat.com. See [attributes L17](https://github.com/theforeman/foreman-documentation/blame/master/guides/common/attributes.adoc#L17)

Reviewer: He should just be using attributes L17 rather than adding {AccessRedHatComVersion} in this case. That is for Red Hat URLs that are not upstream guides, if I understand correctly.

Reviewer 2: My bad. If the content is available upstream, then yes. It's probably something like `{InstallingProjectDocURL}`.

.Example HTTP headers

To pass arguments into a shellhook script, create the following HTTP headers:

[options="nowrap" subs="+quotes,attributes"]
----
{
  "X-Shellhook-Arg-1": "<%= @object.content_view_version_id %>,
  "X-Shellhook-Arg-2": "<%= @object.content_view_name %>
}
----

> Ensure the content renders to a valid JSON. Also, only pass safe fields like database ID, name or labels which do not include new lines or quote characters.

Reviewer: PLease use one sentence per line.
